### PR TITLE
add support for llvm.bitreverse and llvm.bswap

### PR DESCRIPTION
**Pull request by pranavk**
https://llvm.org/docs/LangRef.html#llvm-bitreverse-intrinsics

### CODE REVIEW

File: smt/expr.cpp [hunk 1]
```diff
@@ -523,6 +523,18 @@ expr expr::lshr_exact(const expr &rhs) const {
   return (lshr(rhs) << rhs) == *this;
 }
 
+expr expr::bitreverse() const {
+  C();
+  auto nbits = bits();
+
+  expr res = mkUInt(0, nbits);
+  for (unsigned i = 0; i < nbits; i++) {
+    res = extract(i, i).zext(nbits - 1) | (res << mkUInt(1, nbits));
```
Comment (regehr) [inline smt/expr.cpp]: Pranav, did you investigate any other encodings for this function to see which generate less trouble for the solver?

Comment (nunoplopes) [reply smt/expr.cpp]: I suggest extract of each bit + concat to append the bits together.

File: ir/instr.cpp [hunk 2]
```diff
@@ -254,6 +254,7 @@ void UnaryOp::print(ostream &os) const {
   const char *str = nullptr;
   switch (op) {
   case BitReverse:  str = "bitreverse "; break;
+  case BSwap:       str = "bswap "; break;
```
Comment (LebedevRI) [inline ir/instr.cpp]: The PR title is misleading then?

Comment (pranavk) [reply ir/instr.cpp]: yes, i was about to change that; you were too quick :)

File: smt/expr.cpp (hunk 1 above)
Comment (pranavk) [reply smt/expr.cpp]: i will take care of this in my next push

Comment (pranavk) [reply smt/expr.cpp]: changed to extract + concat

**Pull request merged**